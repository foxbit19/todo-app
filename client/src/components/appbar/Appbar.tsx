import React from 'react'
import { AppBar, Box, Button, Container, Grid, Toolbar, Typography } from '@mui/material'

interface Props {
    onShowCompletedClick: () => void
}

const Appbar = (props: Props) => {
    return (
        <AppBar position="static" enableColorOnDark>
            <Toolbar>
                <Typography variant="h5" style={{ color: 'black', fontFamily: 'Satisfy' }} fontWeight='bold' flexGrow={1}>
                    Easy to do
                </Typography>
                <Button data-testid='show_completed_button' sx={{ color: 'black', marginLeft: 5 }} onClick={props.onShowCompletedClick}>
                    show completed
                </Button>
            </Toolbar>
        </AppBar>
    )
}

export default Appbar