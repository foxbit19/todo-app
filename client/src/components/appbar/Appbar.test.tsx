import React from 'react';
import { render, screen } from '@testing-library/react';
import Item from '../../models/item';
import sinon from 'sinon';
import Appbar from './Appbar';
import userEvent from '@testing-library/user-event';

describe('app bar list', () => {
    test('it matches snapshot', () => {
        const view = render(<Appbar onShowCompletedClick={() => { }} />);
        expect(view).toMatchSnapshot();
    });

    test('it calls the callback when show completed todo is clicked', () => {
        const handleClick = sinon.spy()
        render(<Appbar onShowCompletedClick={handleClick} />);

        userEvent.click(screen.getByTestId('show_completed_button'))

        expect(handleClick.called).toBeTruthy();
    });
})