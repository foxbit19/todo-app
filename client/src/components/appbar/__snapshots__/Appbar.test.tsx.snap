// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app bar list it matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-9t6x5y-MuiTypography-root"
            style="color: black; font-family: Satisfy;"
          >
            Easy to do
          </h5>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-snf0xm-MuiButtonBase-root-MuiButton-root"
            data-testid="show_completed_button"
            tabindex="0"
            type="button"
          >
            show completed
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-9t6x5y-MuiTypography-root"
          style="color: black; font-family: Satisfy;"
        >
          Easy to do
        </h5>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-snf0xm-MuiButtonBase-root-MuiButton-root"
          data-testid="show_completed_button"
          tabindex="0"
          type="button"
        >
          show completed
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
